### PR TITLE
fix: optimize CancelAllOrders gas usage

### PR DESCRIPTION
**Pull request by hallazzang**
## Description

Since we now have new `IterateOrdersByOrderer` method, we can optimize `CancelAllOrders`.

Closes: #295

### CODE REVIEW

File: x/liquidity/keeper/swap_test.go [hunk 1]
```diff
@@ -399,6 +399,31 @@ func (s *KeeperTestSuite) TestCancelAllOrders() {
...
+
+	// The orderer makes an order.
+	orderer := s.addr(1001)
+	s.sellLimitOrder(orderer, pair.Id, utils.ParseDec("1.1"), sdk.NewInt(10000), time.Minute, true)
+
+	// New batch begins, now the orderer can cancel his/her order.
+	liquidity.EndBlocker(s.ctx, s.keeper)
+	liquidity.BeginBlocker(s.ctx, s.keeper)
+
+	s.ctx = s.ctx.WithGasMeter(sdk.NewInfiniteGasMeter()) // to record gas consumption
```
Comment (kogisin) [inline x/liquidity/keeper/swap_test.go:422]: Is this line for initializing consumed gas to 0? If so, don't you think the function name is a bit misleading? This is neither to provide suggestion or feedback; it is rather asking your opinion.

**Review by kogisin — APPROVED** [review]
LGTM, but I feel curious about the `sdk.NewInfiniteGasMeter()` whether or not it is used for initializing consumed gas.

File: x/liquidity/keeper/swap_test.go (hunk 1 above)
Comment (hallazzang) [reply x/liquidity/keeper/swap_test.go:422]: `NewInfiniteGasMeter` is used here to measure gas consumption, without limit. The version with limit is: `NewGasMeter(limit)`

File: x/liquidity/keeper/swap.go [hunk 2]
```diff
@@ -260,43 +260,37 @@ func (k Keeper) CancelOrder(ctx sdk.Context, msg *types.MsgCancelOrder) error {
...
-			canceledOrderIds = append(canceledOrderIds, strconv.FormatUint(order.Id, 10))
+	pairIdSet := map[uint64]struct{}{} // set of pairs where to cancel orders
+	var pairIds []string               // needed to emit an event
+	for _, pairId := range msg.PairIds {
+		if _, found := k.GetPair(ctx, pairId); !found { // check if the pair exists
+			return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "pair %d not found", pairId)
 		}
-		return false, nil
+		pairIdSet[pairId] = struct{}{} // add pair id to the set
+		pairIds = append(pairIds, strconv.FormatUint(pairId, 10))
```
Comment (dongsam) [inline x/liquidity/keeper/swap.go:270]: If cache `orderPairCache` here also, Wouldn't it be possible to increase the cache hit ratio?

Comment (hallazzang) [reply x/liquidity/keeper/swap.go:272]: Thanks, applied the change.

**Review by dongsam — APPROVED** [review]

**Pull request merged**